tests/Install-DbaMultiTool.Tests.ps1

```diff
@@ -0,0 +1,131 @@
+    Context "Validate parameters" {
+        [array]$params = ([Management.Automation.CommandMetaData]$ExecutionContext.SessionState.InvokeCommand.GetCommand($CommandName, 'Function')).Parameters.Keys
+        [object[]]$knownParameters = 'SqlInstance', 'SqlCredential', 'Branch', 'Database', 'LocalFile', 'Force', 'EnableException'
+
+        It "Should only contain our specific parameters" {
+            Compare-Object -ReferenceObject $knownParameters -DifferenceObject $params | Should -BeNullOrEmpty
+        }
+    }
+}
+Describe "$CommandName Integration Tests" -Tags "IntegrationTests" {
+    Context "Testing DBA MultiTool installer with download" {
+        BeforeAll {
+            $branch = "master"
+            $database = "dbatoolsci_multitool_$(Get-Random)"
+            $server = Connect-DbaInstance -SqlInstance $script:instance2
+            $server.Query("CREATE DATABASE $database")
+
+            $resultsDownload = Install-DbaMultiTool -SqlInstance $script:instance2 -Database $database -Branch $branch -Force -Verbose:$false
+        }
+        AfterAll {
+            Remove-DbaDatabase -SqlInstance $script:instance2 -Database $database -Confirm:$false
+        }
+
+        It "Installs to specified database: $database" {
+            $resultsDownload[0].Database -eq $database | Should Be $true
+        }
+        It "Shows status of Installed" {
+            $resultsDownload[0].Status -eq "Installed" | Should Be $true
+        }
+        It "Installed sp_doc, sp_helpme, sp_sizeoptimiser, and sp_estindex" {
+            'sp_doc', 'sp_helpme', 'sp_sizeoptimiser', 'sp_estindex' | Should BeIn $resultsDownload.Name
+        }
+        It "Has the correct properties" {
+            $result = $resultsDownload[0]
+            $ExpectedProps = 'SqlInstance,InstanceName,ComputerName,Name,Status,Database'.Split(',')
+            ($result.PsObject.Properties.Name | Sort-Object) | Should Be ($ExpectedProps | Sort-Object)
+        }
+        It "Shows status of Updated" {
+            $resultsDownload = Install-DbaMultiTool -SqlInstance $script:instance2 -Database $database -Verbose:$false
+            $resultsDownload[0].Status -eq 'Updated' | Should -Be $true
+        }
+        It "Shows status of Error" {
+            $folder = Join-Path (Get-DbatoolsConfigValue -FullName Path.DbatoolsData) -Child "DBA-MultiTool-$branch"
+            $sqlScript = (Get-ChildItem $folder | Select-Object -First 1).FullName
+            Add-Content $sqlScript (New-Guid).ToString()
+            $result = Install-DbaMultiTool -SqlInstance $script:instance2 -Database $database -Verbose:$false
+            $result[0].Status -eq "Error" | Should -Be $true
+        }
+    }
+    Context "Testing DBA MultiTool installer with LocalFile" {
+        BeforeAll {
+            $branch = "master"
+            $database = "dbatoolsci_multitool_$(Get-Random)"
+            $server = Connect-DbaInstance -SqlInstance $script:instance3
```

Comment:
Please use the same instance across your test.

The AfterAll cleans up the objects created from within the last context so the use of the same instance in each context should not cause any contention.

Reply:
Using all three instances we have built for Appveyor is not necessary for this command.